typesense/src/client/mod.rs

```diff
@@ -309,7 +308,7 @@ impl Client {
         };
 
         for _ in 0..self.nodes.len() {
-            let node = self.nodes.get_safe_unchecked(index);
+            let node = &self.nodes[index];
```

Comment:
Can you check if it has the same assembly? Because usually the compiler leaves a panic handler there, which makes the code to be less optimal.